### PR TITLE
Add ability to reorient the UI while lying down

### LINKED ISSUE #544: Add ability to rotate the screen up while lying down
## Hardware
Oculus Go

## Steps to Reproduce
1. Start browser
2. Lie down in bed / recline
3. Recenter screen

## Current Behavior
Everything stays as if you are seated/standing

## Expected Behavior
Able to move the browser or recenter above your face

### PR DESCRIPTION
**Pull request by MortimerGoro**
Fixes #544

### CODE REVIEW

**Comment by MortimerGoro** [comment]
- Updated the PR with bigger threshold for looking down.
- Fixed the Wave left/right reorientation.

File: app/src/main/cpp/DeviceDelegate.h [hunk 1]
```diff
@@ -54,6 +55,8 @@ class DeviceDelegate {
   virtual void StartFrame() = 0;
   virtual void BindEye(const device::Eye aWhich) = 0;
   virtual void EndFrame(bool aDiscard = false) = 0;
+
+  static vrb::Matrix CalculateReorientationMatrix(const vrb::Matrix& aHeadTransform, const vrb::Vector& aHeightPosition);
```
[the remark this answers is not in the document]
Comment (MortimerGoro) [reply app/src/main/cpp/DeviceDelegate.h]: I created a new DeviceUtils class in the latest PR.

**Comment by avrignaud** [comment]
This makes me think we should get a process in place to collect release notes, and make sure we accidentally don't drop anything interesting/useful to announce. @MortimerGoro  @bluemarvin do you know if there's a GH way to collect these (other than searching a huge number of closed bugs)? Or maybe we should just create a Gdoc for the next version and collect there?

**Comment by keianhzo** [comment]
@avrignaud maybe a good place for that is: https://github.com/MozillaReality/FirefoxReality/releases

**Comment by bluemarvin** [comment]
> This makes me think we should get a process in place to collect release notes, and make sure we accidentally don't drop anything interesting/useful to announce. @MortimerGoro @bluemarvin do you know if there's a GH way to collect these (other than searching a huge number of closed bugs)? Or maybe we should just create a Gdoc for the next version and collect there?

Github has a compare feature:

[https://github.com/MozillaReality/FirefoxReality/compare/1.0-rc1...master](https://github.com/MozillaReality/FirefoxReality/compare/1.0-rc1...master)

That link shows all changes that have landed since the 1.0a release. Once we cut a 1.0.1 release:

[NOTE: This link won't work until we cut th 1.0.1 release]: [https://github.com/MozillaReality/FirefoxReality/compare/1.0a...1.0.1](https://github.com/MozillaReality/FirefoxReality/compare/1.0a...1.0.1) will show all commits that have landed since 1.0a release.

**Comment by bluemarvin** [comment]
I guess I'm confused what the desired behavior is for this feature. Should the window only be rotated about the X axis or are is rotation about z-axis also desired. I guess for lying on your side?

**Comment by bluemarvin** [comment]
I wonder if there should be a setting to turn this off?

**Comment by cvan** [comment]
actually, after doing some further testing, I realised the orientation reset seems to be askew. try opening the keyboard, adjusting the orientation, and try typing. that's probably the most obvious STR I've found. but it definitely feels off.

**Comment by avrignaud** [comment]
@thenadj Think it might be good to get a design together for keyboard/screen positioning - something that gives the user explicit control, and that we can focus test in future to understand best default starting position.

**Comment by MortimerGoro** [comment]
z-axis scale is for laying on your side. Oculus Go browser also allows it.

We could add a UI setting to enable/disable this. What do you think @avrignaud @thenadj ?

**Comment by cvan** [comment]
I would suggest the implementation be consistent with Oculus Browser's. I don't think it needs to be any more complicated than that.

**Comment by bluemarvin** [comment]
This seems to work exactly like the oculus browser. Let's land and add any UI in a follow up.

**Comment by thenadj** [comment]
Design for this is on the to do list.

**Pull request merged**